app/models/katello/host/subscription_facet.rb

```diff
@@ -34,6 +35,19 @@ def import_database_attributes(consumer_params)
         end
       end
 
+      def update_consumer_type(consumer_params)
+        if consumer_params.try(:[], 'type').try(:[], 'label') == 'hypervisor'
+          self.consumer_type = 'hypervisor'
+        elsif consumer_params.try(:[], 'facts').try(:[], 'virt.is_guest') == "true"
+          self.consumer_type = 'guest'
+        elsif !consumer_params.try(:[], 'guestIds').empty?
```

Comment:
just wondering why this is not in a fact parser, similar to how we parse other facts.